nav2_bt_navigator/src/navigators/navigate_through_poses.cpp

```diff
@@ -42,6 +42,13 @@ NavigateThroughPosesNavigator::configure(
 
   path_blackboard_id_ = node->get_parameter("path_blackboard_id").as_string();
 
+  // Get the action server client node
+  rclcpp::Node::SharedPtr client_node = bt_action_server_->getBlackboard()->get<rclcpp::Node::SharedPtr>("node");
```

Comment:
Where is this node inserted? Would it not be possible to do this around there instead to group blackboard operations together?

Reply:
This node is inserted in the bt_action_server_impl.hpp file in Line 149. I think I could not directly access it, as the client_node_ in bt_action_server.hpp is protected. You would need to add a function to directly access it.

Reply:
https://github.com/ros-planning/navigation2/blob/78c693a5d29da86d696ee535e303897220be4663/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp#L123

The client node is created at the top of that function - you have direct access to it

Reply:
I added the parameters directly in the bt_action_server_impl.hpp.